test/podFailover/podFailoverCrd/podfailure.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by client-gen. DO NOT EDIT.
```

Comment:
How was this file generated? Usually, the CRD types are defined in `types.go` and decorated with `// +genclient` annotations. This looks like the base types file which should be the source, but I don't see the annotations.

It would be good to document the process for generating these CRDs in case someone needs to update them in the future.

Reply:
I wrote all of this by hand mostly. I had to add the comments at the top to pass the verify scripts but maybe I should generate them. I had just written all the CRD code before realizing there was a verify script

Reply:
Yeah, it would be good to use the code generator to generate the client code and CRD definitions.

Reply:
I am OK with deferring the code generator change to a subsequent PR.